### PR TITLE
Js memoize

### PR DESCRIPTION
**Pull request by jsingle**
@thieman MEMOIZE

### CODE REVIEW

File: dusty/memoize.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import functools
+import pickle
+
+from .payload import function_key
+
+cache = {}
+
+def memoized(fn):
+    '''
```
Comment (thieman) [inline dusty/memoize.py]: """

File: tests/testcases.py [hunk 2]
```diff
@@ -49,12 +50,14 @@ def setUp(self):
...
         self.capture_handler = TestCaptureHandler(self.client_output)
         logging.getLogger(constants.SOCKET_LOGGER_NAME).addHandler(self.capture_handler)
+        reset_memoize_cache()
 
     def tearDown(self):
         os.remove(self.temp_config_path)
         shutil.rmtree(self.temp_specs_path)
         shutil.rmtree(self.temp_repos_path)
         logging.getLogger(constants.SOCKET_LOGGER_NAME).removeHandler(self.capture_handler)
+        reset_memoize_cache()
```
Comment (thieman) [inline tests/testcases.py]: Can't we get away with just doing this at the end of each tearDown, not also during setUp?

File: tests/unit/memoize_test.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    def setUp(self):
+        super(TestMemoize, self).setUp()
+        self.counter = 0
+        @memoized
+        def memoized_fn():
+            self.counter += 1
+        self.memoized_fn = memoized_fn
+
+    def test_memoize(self):
+        self.memoized_fn()
```
Comment (thieman) [inline tests/unit/memoize_test.py]: Probably worth adding a test to make sure the function is actually returning what you think it should be

File: tests/testcases.py (hunk 2 above)
Comment (jsingle) [inline tests/testcases.py]: nope :)

File: tests/unit/memoize_test.py (hunk 3 above)
Comment (jsingle) [inline tests/unit/memoize_test.py]: +1

File: dusty/memoize.py (hunk 1 above)
Comment (jsingle) [inline dusty/memoize.py]: Haha I WAS COPYING GCLIB

File: tests/testcases.py (hunk 2 above)
Comment (thieman) [inline tests/testcases.py]: :anguished:

Comment (jsingle) [inline tests/testcases.py]: Yeah not sure why but that's what i observed. Didn't feel like digging into why

**Pull request merged**